.github/workflows/_build-and-push-cliain.yml

```diff
@@ -0,0 +1,120 @@
+---
+name: Build/Check and optionally push cliain docker image
+
+on:
+  workflow_call:
+    inputs:
+      liminal-runtime:
+        description: 'Set to true if cliain should be built with liminal runtime'
+        required: true
+        type: boolean
+        default: false
+      check-only:
+        description: 'Set to true if compilation linking phase should be omitted'
+        required: true
+        type: boolean
+        default: false
+
+jobs:
+  build-or-check-cliain:
+    name: Build or check cliain binary
+    runs-on: [self-hosted, Linux, X64, large]
+    env:
+      CARGO_COMMAND: ${{ inputs.check-only && 'check' || 'build' }}
+      FEATURES: ${{ inputs.liminal-runtime && '--features liminal' || '' }}
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: Cardinal-Cryptography/github-actions/install-rust-toolchain@v4
+
+      - name: ${{ env.CARGO_COMMAND }} cliain binary
+        run: |
+          cd ./bin/cliain && cargo ${{ env.CARGO_COMMAND }} --release ${{ env.FEATURES }}
+
+      - name: Test SNARK relations
+        if: ${{ inputs.liminal-runtime && inputs.check-only != true }}
+        run: ./.github/scripts/test_cliain_snark_relations.sh
+
+      - name: Upload cliain binary to GH artifacts
+        if: ${{ inputs.check-only != true }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: cliain
+          path: bin/cliain/target/release/cliain
+          if-no-files-found: error
+          retention-days: 7
+
+  push-cliain-to-ecr:
+    name: Build and push cliain docker image to ecr
+    needs: [build-or-check-cliain]
+    if: ${{ inputs.check-only != true }}
```

Comment:
does it mean that we cannot just `build` and run `test_cliain_snark_relations.sh`?

Reply:
sorry, my bad; `check` is sufficient for Ci; my real concern is about `test_cliain_snark_relations.sh` (and for that we actually need to `build`); is it possible to run this script as normal tests, i.e. for PRs, but not for merge? if I understand PR code correctly, this is tight to pushing and thus is run only for merges

Reply:
Yes but this script requires binary `cliain`, which one we should use then? For PR, `clian` is not built, just checked